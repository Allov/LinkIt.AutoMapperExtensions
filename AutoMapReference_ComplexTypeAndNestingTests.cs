using ApprovalTests.Reporters;
using AutoMapper;
using NUnit.Framework;
using RC.Testing;

namespace ShowMeAnExampleOfAutomapperFromLinkedSource
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class AutoMapReference_ComplexTypeAndNestingTests {

        [SetUp]
        public void SetUp()
        {
            Mapper
                .CreateMap<NestedLinkedSource, MyComplexDto>()
                .MapModel<NestedLinkedSource, MyComplexModel, MyComplexDto>();
            Mapper.CreateMap<MyPoint, MyPointDto>();
        }

        [Test]
        public void AssertConfigurationIsValid()
        {
            Mapper.AssertConfigurationIsValid();
        }

        [Test]
        public void MapModelOnly_WithIntAndComplexType_ShouldMap()
        {
            var source = CreateMyComplexModel(1, null);

            var actual = Mapper.Map<MyComplexDto>(source);

            ApprovalsExt.VerifyPublicProperties(actual);
        }

        private static MyComplexModel CreateMyComplexModel(int id, int? childId)
        {
            return new MyComplexModel{
                Id = id,
                Title = "Title "+id,
                Point = new MyPoint {X = id, Y = id},
                ChildId = childId
            };
        }

        public class NestedLinkedSource:ILinkedSource<MyComplexModel>
        {
            public MyComplexModel Model { get; set; }
            public NestedLinkedSource Child { get; set; }
        }

        public class MyComplexModel {
            public int Id { get; set; }
            public string Title { get; set; }
            public int? ChildId { get; set; }
            public MyPoint Point { get; set; }
        }

        public class MyPoint {
            public float X { get; set; }
            public float Y { get; set; }
        }

        public class MyComplexDto {
            public int Id { get; set; }
            public int Title { get; set; }
            public MyComplexDto Child { get; set; }
            public MyPointDto Point { get; set; }
        }

        public class MyPointDto {
            public float X { get; set; }
            public float Y { get; set; }
        }


    }
}