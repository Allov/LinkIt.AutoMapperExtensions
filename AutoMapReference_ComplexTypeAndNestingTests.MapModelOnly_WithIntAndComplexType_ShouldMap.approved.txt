{
  "id": 1,
  "title": "Title 1",
  "child": null,
  "point": {
    "x": 1.0,
    "y": 1.0
  }
}